### ISSUE TITLE
New Glossary: Cross-Origin isolation

### ISSUE THREAD

**Issue by skyclouds2001** [mdn/content#37287]
### MDN URL

none

### What information was incorrect, unhelpful, or incomplete?

add a glossary for Cross-Origin isolation

### What did you expect to see?

same as above

### Do you have any supporting links, references, or citations?

https://blog.stackblitz.com/posts/cross-browser-with-coop-coep/ a link for reference

**Comment by hamishwillee** [comment]
I'm somewhat supportive of this idea. If we do this it should take much of the information from https://developer.mozilla.org/en-US/docs/Web/API/Window/crossOriginIsolated , and link that and the Worker equivalent. It should also link the https://blog.stackblitz.com/posts/cross-browser-with-coop-coep/ , which while a little out of data w.r.t. browser support is a really good overview of the practical implications of all this stuff for real users at scale.

My only concern is what we leave in the `Window.crossOriginIsolated` topic. Duplication is not idea. Also note that many of the places that link to  `Window.crossOriginIsolated` would now link to the glossary.

**Comment by wbamberg** [comment]
I am planning to write something long-form about this whole issue (xs leaks, Spectre, COOP, COEP etc). I'm not that convinced a glossary entry is quite the right vehicle as it's kind of a complicated topic. But since I'm not going to get around to this any time soon, so I'm not going to stop anyone writing a glossary entry in the meantime.

(In general I'm not sure when we should decide to make something a glossary entry or something else. I think the ideal glossary entry is something pretty short that really defines some piece of technical jargon, especially then it's cross-technology. Sometimes I think we have glossary entries that ought to be longer-form, and e.g. I just deleted the "clickjacking" glossary entry in favour of a guide page, and will probably try to do the same with the XSS page. But maybe I shouldn't? Maybe we should have short glossary pages that point to long-form explanations? I'm not sure.)

**Comment by Josh-Cena** [comment]
I agree Glossary should only be for concepts that cannot be adequately placed anywhere else. For example, "ASCII", "Git", or "OOP" are good candidates. However, we should prefer to colocate concepts with their references wherever possible to avoid making glossary bloat up infinitely.

**Comment by hamishwillee** [comment]
Those are reasonable views to take, and philosophically I agree. 

Pragmatically I think it would be good now to reduce the duplication, and a glossary topic is a reasonable solution for that. When/if a longer form topic exists in the HTTP space for this, it could replace such a glossary topic - provided it still provides a concise link end point for what cross-origin isolation means.

Or we could wait.

**Comment by Josh-Cena** [comment]
I would think `crossOriginIsolated` is a good enough source to link to, personally.

**Comment by skyclouds2001** [comment]
> I would think `crossOriginIsolated` is a good enough source to link to, personally.

But note it is two pages: [`Window.crossOriginIsolated`](https://developer.mozilla.org/en-US/docs/Web/API/Window/crossOriginIsolated) and [`WorkerGlobalScope.crossOriginIsolated`](https://developer.mozilla.org/en-US/docs/Web/API/WorkerGlobalScope/crossOriginIsolated), where infomation are duplicated between the two properties.

The cross-origin isolation glossary page can store info about how to be cross-origin isolation and features managed by cross-origin isolation, etc. Then those info  could remove from the two pages and simply and a link to the glossary page. So this can reduce the duplicated info.

see also https://github.com/mdn/content/issues/6856

**Comment by Josh-Cena** [comment]
> But note it is two pages: [`Window.crossOriginIsolated`](https://developer.mozilla.org/en-US/docs/Web/API/Window/crossOriginIsolated) and [`WorkerGlobalScope.crossOriginIsolated`](https://developer.mozilla.org/en-US/docs/Web/API/WorkerGlobalScope/crossOriginIsolated), where infomation are duplicated between the two properties.

It doesn't have to. Many `Window` pages are more complete than `WorkerGlobalScope` counterparts. We can simply say "for more information, see the `Window` page instead"; I doubt people are reading the `WorkerGlobalScope` ones without checking the `Window` ones anyway.

**Comment by wbamberg** [comment]
Some of the conversation in https://github.com/mdn/content/issues/3526 is relevant to this.

**Comment by hamishwillee** [comment]
> Some of the conversation in [#3526](https://github.com/mdn/content/issues/3526) is relevant to this.

Yes. You may (or may not) have missed the bit about "Origin Agent Cluster header". In addition to this concept of cross-origin isolation my understanding is that there are other groupings of resources related to origins - such as the "agent cluster". We don't capture those, and perhaps we don't need to. But if we do, then it kind of implies we might need a bigger/guide page around all of these concepts related to isolating and severing relationships.

**Comment by Josh-Cena** [comment]
By the way, the agent cluster formalism is now properly introduced in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Execution_model#agent_clusters_and_memory_sharing, because it is a JS core concept. You might want to link to it in HTTP or API.

**Comment by hamishwillee** [comment]
> By the way, the agent cluster formalism is now properly introduced in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Execution_model#agent_clusters_and_memory_sharing, because it is a JS core concept. You might want to link to it in HTTP or API.

Wow. Well done. Much clearer (than I recall) from scan-reading the spec.
